force-app/main/default/triggers/OpportunityTrigger.trigger

```diff
@@ -0,0 +1,94 @@
+trigger OpportunityTrigger on Opportunity (before update,after update, before delete) {
+
+  // Prevent recursive updates
+  if (OpportunityTriggerHandler.isTriggerExecuted) {
+      return;
+  }
+  OpportunityTriggerHandler.isTriggerExecuted = true;
+
+  //Before update
+  if (Trigger.isBefore && Trigger.isUpdate) {
+
+    for (Opportunity opp : Trigger.new){
+      if (opp.Amount <=5000) {
+        opp.addError('Opportunity amount must be greater than 5000');
+      }
+    }
+
+  }
+
+  // Before Delete
+  if (Trigger.isBefore && Trigger.isDelete){
+    Set<Id> accountId = new Set<Id>();
+    
+    for(Opportunity opp : Trigger.old) {
+      if (opp.StageName == 'Closed Won' &&
+          opp.AccountId != null){
+
+            accountId.add(opp.AccountId);
+          }
+    }
+
+    //query the accounts with the matching Id from the set
+    Map<Id, Account> accountMap = new Map<Id, Account>([SELECT Id, Industry 
+                                                      FROM Account WHERE Id IN :accountId ]);
+
+    // iterate  through the opportunities to check the conditions before deleting
+    for (Opportunity opp : Trigger.old) {
+      if(opp.StageName == 'Closed Won') {
+        Account relatedAccount = accountMap.get(opp.AccountId);
+        if (relatedAccount !=null && relatedAccount.Industry == 'Banking'){
+          // prevent deletion and add an error message 
+          opp.addError('Cannot delete closed opportunity for a banking account that is won');
+        }
+      }
+    }
+  }
+
+  //after update
+
+  if(Trigger.isAfter && Trigger.isUpdate){
```

Comment:
Try to see if this can be done in the before context. If so, you can avoid the DML at the end